### PR TITLE
Fix default dtype in color twist family of operators

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

## Category:

**Bug fix**

## Description:

- Fixes https://github.com/NVIDIA/DALI/issues/4065
- Default dtype was wrongly set to UINT8 instead of NO_TYPE, which caused the input type inference mechanism not to work properly.
- The PR fixes that and adds a test

### Affected modules and functionalities:

Color twist family of operators

### Key points relevant for the review:

NA

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/image/color/color_twist.cc [hunk 1]
```diff
@@ -59,7 +59,7 @@ DALI_SCHEMA(ColorTransformBase)
     .AddOptionalTypeArg(color::kOutputType, R"code(Output data type.
 
 If not set, the input type is used.)code",
-                    DALI_UINT8)
+                        DALI_NO_TYPE)
```
Comment (mzient) [inline dali/operators/image/color/color_twist.cc:62]: Why not just remove it?

Comment (jantonguirao) [reply dali/operators/image/color/color_twist.cc:62]: sure, I can remove it

**Comment by mzient** [comment]
I've removed the "important fix" label, because it's also a breaking change.

**Review by prak-nv — APPROVED** [review]

**Review by mzient — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Pull request merged**